### ISSUE TITLE
app-id nil

### ISSUE THREAD

**Issue by chrismurrph** [fulcrologic/fulcro-inspect#18]
I am getting this 3 times when starting my application that uses the tool:

`[  2.547s] [fulcro.client] Mutation fulcro.inspect.ui.transactions/add-tx failed with exception Error: Assert failed: Path [:fulcro.inspect.ui.transactions/tx-list-id [:fulcro.inspect.core/app-id nil] :fulcro.inspect.ui.transactions/tx-list] for append must target an app-state vector.
(vector? (get-in state data-path))`

Does the host application have to have a "displayName"? From the `fulcro.inspector` source:

`(some-> reconciler fp/app-root fp/react-type (gobj/get "displayName") symbol)`

What is this "displayName" and how to ensure that hosting applications always have it?

**Comment by wilkerlucio** [comment]
@chrismurrph this is usually defined by every react component, most tools use that to check the component name, `defui` automatically attaches this data on the component definition. Are you using some non-fulcro component as your root? Another question, are you compiling with `shadow-cljs`?

**Comment by chrismurrph** [comment]
My *root* is a standard one that starts off: `defui ^:once Root`. Not using `shadow-cljs` (don't know what that is, have just heard it mentioned). I suppose I should try to prove that there is in fact no "displayName" on my root component, using that `(some-> ...)` code above. Will try to do that and get back to you. Anything else I can do?

**Comment by wilkerlucio** [comment]
Ok, thanks for the updates, I'm not sure what's going on in this case. One thing you can do is force your app to have a pre-defined `::app-id`, to do that, add this to your root initial state: `::f.inspect/app-id "My App"`. But it worth investigation, is this project you are working on public?

**Comment by chrismurrph** [comment]
I have been able to prove the opposite, that in fact "displayName" is on the root component. I will put some asserts in and install your library locally in the .m2 repo and see what I can find out. Give me a day or two... Yes I can see explicitly putting in the app-id as a possible work around, as you do for the tools devcards. But also that install sets the app-id, so that should not be necessary in my case, and in fact I don't think it helps because I've already tried that approach. Not in public, but I have full control over it.

**Comment by wilkerlucio** [comment]
@chrismurrph Hey, I just stumbled on the same problem. I did some debugging here; the problem is that Fulcro is triggering transactions before the `:started-callback` is called, so before inspect had a chance to initialize your app, this causes the app to no be found and trigger the error. When you set the state upfront it can use that to save the information even with the app non-initialized. I'm going to just ignore those transactions before the app is ready, they are just related to initial data normalization, safe to ignore.
